TarponSimulator2017/Core/Fish.cs

```diff
@@ -20,6 +22,32 @@ public Fish (int StartAbscisse, int StartOrdinate) : base (FrictionForceFish, Ac
 			this.CurrentState = new FishStateTriggered (this);
 		}
 
+		/// <summary>
+		/// Creates the random fish.
+		/// </summary>
+		/// <returns>The random fish.</returns>
+
+
+		/// <summary>
+		/// Changes the state to caught the hook if allowed.
+		/// </summary>
+		public void ChangeStateToCaughtTheHook ()
+		{
+			if (IsStateTransitionAllowed (typeof(FishStateCaughtTheHook))) {
+				this.CurrentState = new FishStateCaughtTheHook (this);
+			}
+		}
+
+		/// <summary>
+		/// Determines whether this instance can set its current state to desired state
+		/// </summary>
+		/// <returns><c>true</c> if this instance can set is current state to desired state; otherwise, <c>false</c>.</returns>
+		/// <param name="TypeOfNextState">Type of next state.</param>
+		public Boolean IsStateTransitionAllowed (Type TypeOfNextState)
```

Comment:
The beginning of a state machine? For the future?
I think you'll take the current state and the given state and check in a matrix that the transition is allowed?
Could we extract this logic from Core later?

And an open question: Should we handle states with classes?
Here is how I implemented a state machine in Javascript, and like how you name transitions, and afterward attach some callbacks to transitions: https://github.com/superboum/quantum/blob/master/public/js/foundation/state_machine.js

Edit: after reading more, I think we will need classes, but we need a discussion on how we handle states, but even more important, transitions (look at the javascript code if you don't see what I mean).